src/src/test/java/StringManipulationTest.java

```diff
@@ -0,0 +1,183 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StringManipulationTest {
+
+    private StringManipulationInterface manipulatedstring;
+
+    @BeforeEach
+    public void setUp() {
+        manipulatedstring = new StringManipulation();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        manipulatedstring = null;
+    }
+
```

Comment:
You could add comments before each test case to explain what each one is verifying.